engine/src/main/java/org/terasology/rendering/cameras/SubmersibleCamera.java

```diff
@@ -0,0 +1,43 @@
+package org.terasology.rendering.cameras;
+
+import org.terasology.config.RenderingConfig;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.rendering.RenderHelper;
+import org.terasology.world.WorldProvider;
+
+public abstract class SubmersibleCamera extends Camera {
+	
+	/* Used for Underwater Checks */
+    private WorldProvider worldProvider;
+    private RenderingConfig renderingConfig;
+
+    public SubmersibleCamera(WorldProvider worldProvider, RenderingConfig renderingConfig) {
+	this.worldProvider = worldProvider;
+    	this.renderingConfig = renderingConfig;
+    }
+	
+   /**
+    * Returns True if the head of the player is underwater. False otherwise.
+    *
+    * Implementations must take in account waves if present.
```

Comment:
Well, this class -is- taking in account waves if present. So, I'd change to:

` * Takes in account waves if present.`